src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,17 +30,98 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         this.isPrivate = isPrivate;
-        if (!isValidAddress(address)) {
-            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+        try {
+            partitionAddress(address);
+        } catch (IllegalValueException e) {
+            throw e;
         }
+        
         this.value = address;
     }
 
     /**
      * Returns true if a given string is a valid person email.
      */
-    public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+    public static boolean isValidAddress(String address) throws IllegalValueException{
+        try {
+            HashMap<String, String> addressParams = extractAddressParam(address);
+            String block = addressParams.get("block");
+            String street = addressParams.get("street");
+            String unit = addressParams.get("unit");
+            String postalCode = addressParams.get("postalCode");
+            
+            if (!Block.isValidBlock(block)) {
+                throw new IllegalValueException(Block.MESSAGE_BLOCK_CONSTRAINTS);
+            } else if (!Street.isValidStreet(street)) {
+                throw new IllegalValueException(Street.MESSAGE_STREET_CONSTRAINTS);
+            } else if (!Unit.isValidUnit(unit)) {
+                throw new IllegalValueException(Unit.MESSAGE_UNIT_CONSTRAINTS);
+            } else if (!PostalCode.isValidPostalCode(postalCode)) {
+                throw new IllegalValueException(PostalCode.MESSAGE_POSTAL_CODE_CONSTRAINTS);
+            }
+        } catch (IllegalValueException e) {
+            throw e;
+        }
+        return true;
+    }
+    
+    /**
+     * Extract the different parts of the address
+     * 
+     * @return HashMap containing the different parts of the address
+     */
+    private static HashMap<String, String> extractAddressParam(String address)throws IllegalValueException {
+        StringTokenizer st = new StringTokenizer(address, ",");
+        String[] addressParamArr = new String[ADDRESS_PARAM_COUNT];
+        int count=0;
+        while (st.hasMoreTokens()) {
+            addressParamArr[count++] = st.nextToken().trim();
+        }
+        
+        if (count == ADDRESS_PARAM_COUNT) {
+            HashMap<String, String> addressParamHashMap = new HashMap<String, String>();
+            addressParamHashMap.put("block", addressParamArr[0]);
```

Comment:
if i say, try to avoid magic numbers here, would it be asking for too much :-D

i appreciate the amount of effort that has gone into this commit, it is quite extensive :+1: